packages/ffe-cards-react/src/components/WithCardAction.tsx

```diff
@@ -9,30 +16,52 @@ export type WithCardActionProps<As extends ElementType = 'div'> = Omit<
     ComponentAsPropParams<As>,
     'children'
 > & {
+    baseClassName: string;
     children: (props: { CardAction: typeof CardAction }) => React.ReactNode;
 };
 
 function WithCardActionForwardRef<As extends ElementType>(
     props: WithCardActionProps<As>,
     ref: ForwardedRef<any>,
 ) {
-    const { children, as: Comp = 'div', onClick, ...rest } = props;
-    const actionInnerRef = useRef<HTMLAnchorElement>(null);
+    const {
+        children,
+        as: Comp = 'div',
+        onClick,
+        className,
+        baseClassName,
+        ...rest
+    } = props;
+    const cardActionInnerRef = useRef<HTMLElement>(null);
+    const [isUsingCardAction, setIsUsingCardAction] = useState<boolean>();
+
+    useEffect(() => {
+        if (cardActionInnerRef.current && isUsingCardAction === undefined) {
+            setIsUsingCardAction(true);
+        }
+    }, [isUsingCardAction]);
```

Comment:
Jeg forstår ikke hvorfor hooken ble endret fra denne:
```js
    useEffect(() => {
        if (cardActionInnerRef.current) {
            setIsUsingCardAction(true);
        }
    }, []);
```
`isUsingCardAction` vil alltid være `undefined` første gang og endres aldri utenfor denne hooken.

Tenkte du å oppdage når CardAction blir lagt til eller fjernet? Da må man vel gjøre noe slikt:

```js
    useEffect(() => {
          setIsUsingCardAction(cardActionInnerRef.current != null);
    }, [cardActionInnerRef.current]);
```

Reply:
Deler up svaret her

siste forslaget 
```
  console.log(cardActionInnerRef.current); /// undefined
```

Parent render ikke på nytt når children rendrer. Det er setState som trigger en render.

Reply:
Således vill denne heller funka eftersom effecten aldrig blir kjørt. 
```
  useEffect(() => {
        setIsUsingCardAction(cardActionInnerRef.current != null);
    }, [cardActionInnerRef.current]);
```

eslint er heller ikke førnøjd med den. 
```
ESLint: React Hook useEffect has an unnecessary dependency: 'cardActionInnerRef.current'. Either exclude it or remove the dependency array. Mutable values like 'cardActionInnerRef.current' aren't valid dependencies because mutating them doesn't re-render the component.(react-hooks/exhaustive-deps)

```

Reply:
Jag må trigge en setState før att rendre komponenten på nytt med den nya klassen,.

Reply:
Men det gir ikke mening å ha `isUsingCardAction`som dependency når `setIsUsingCardAction` kun kalles fra denne effecten. Det betyr bare at effecten vil trigge seg selv på nytt (og nå er isUsingCardAction definert, slik at loopen brytes).
```js
    useEffect(() => {
        if (cardActionInnerRef.current && isUsingCardAction === undefined) {
            setIsUsingCardAction(true);
        }
    }, [isUsingCardAction]);
```

Dette er altså funksjonelt identisk som over:
```js
    useEffect(() => {
        if (cardActionInnerRef.current) {
            setIsUsingCardAction(true);
        }
    }, []);
```

Det stemmer at det ikke vil fungere uten `useEffect()`, fordi refs settes etter at koden har kjørt (men før effects kjøres).

Reply:
Dersom CardAction legges til eller fjernes uten at hele kortet remountes, så vil ikke `isUsingCardAction` oppdateres med koden over. Mulig det kan løses med å depende på `children` slik:
```js
    useEffect(() => {
        setIsUsingCardAction(cardActionInnerRef.current != null);
    }, [children]);
```
Men tenker at det kanskje er et veldig uvanlig tilfelle som vi ikke trenger støtte.

Reply:
Siste forslaget er fint går til det. Endaste grunden til att `isUsingCardAction` var i hooken er lint. Jag vet kanskje ikke blir stoppad her emn er van fra våre løsningar i sparing att man blir det. Brukes det i hoppen skall det vare i dependency array.

Reply:
I min erfaring gir lintere mange falske positiver på `useEffect()` :shrug:

Reply:
Med koden over vil det ikke fungere å ta bort CardAction. Foreslår isåfall følgende:
```js
    useEffect(() => {
        const isStillUsingCardAction = cardActionInnerRef.current != null;
        if (isUsingCardAction != isStillUsingCardAction) {
            setIsUsingCardAction(isStillUsingCardAction);
        }
    });
```

Reply:
`I min erfaring gir lintere mange falske positiver på useEffect() 🤷` det tror jag på.. Blitt van att passa på det.